src/EditorFeatures/CSharpTest/CodeActions/NamedArguments/AddNamedArgumentsTests.cs

```diff
@@ -0,0 +1,141 @@
+index: 0);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddNamedArguments)]
+        public async Task TestThisConstructorInitializer()
+        {
+            await TestAsync(
+@"class C { C(int arg1, int arg2) {} C() : [|this(1, 2)|] {} }",
+@"class C { C(int arg1, int arg2) {} C() : this(arg1: 1, arg2: 2) {} }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddNamedArguments)]
+        public async Task TestBaseConstructorInitializer()
+        {
+            await TestAsync(
+@"class C { public C(int arg1, int arg2) {} } class D : C { D() : [|base(1, 2)|] {} }",
+@"class C { public C(int arg1, int arg2) {} } class D : C { D() : base(arg1: 1, arg2: 2) {} }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddNamedArguments)]
+        public async Task TestConstructor()
+        {
+            await TestAsync(
+@"class C { C(int arg1, int arg2) { new [|C|](1, 2); } }",
+@"class C { C(int arg1, int arg2) { new C(arg1: 1, arg2: 2); } }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddNamedArguments)]
+        public async Task TestIndexer()
+        {
+            await TestAsync(
+@"class C { char M(string arg1) => [|arg1|][0]; }",
+@"class C { char M(string arg1) => arg1[index: 0]; }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddNamedArguments)]
+        public async Task TestArrayIndexer()
+        {
+            await TestMissingAsync(
+@"class C { int M(int[] arg1) => [|arg1|][0]; }");
+
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddNamedArguments)]
+        public async Task TestMissingArgumentNames()
+        {
+            await TestAsync(
+@"class C { void M(int arg1, int arg2) => [|M|](1, 2); }",
+@"class C { void M(int arg1, int arg2) => M(arg1: 1, arg2: 2); }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddNamedArguments)]
+        public async Task TestSingleParams()
+        {
+            await TestAsync(
+@"class C { void M(int arg1, params int[] arg2) => [|M|](1, 2); }",
+@"class C { void M(int arg1, params int[] arg2) => M(arg1: 1, arg2: 2); }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddNamedArguments)]
+        public async Task TestMultipleParams()
+        {
+            await TestAsync(
+@"class C { void M(int arg1, params int[] arg2) => [|M|](1, 2, 3); }",
+@"class C { void M(int arg1, params int[] arg2) => M(arg1: 1, arg2: new int[] { 2, 3 }); }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddNamedArguments)]
+        public async Task TestExistingArgumentNames()
+        {
+            await TestAsync(
+@"class C { void M(int arg1, int arg2) => [|M|](1, arg2: 2); }",
+@"class C { void M(int arg1, int arg2) => M(arg1: 1, arg2: 2); }");
+        }
```

Comment:
@balajikris Not yet, is that OK to suggest fix for diagnostics in a CodeRefactoringProvider?